### PR TITLE
Feature/12/api foundation

### PR DESCRIPTION
**Pull request by EV3RETH**

### CODE REVIEW

File: packages/ts-api/src/routes/example.ts
Comment (erak) [inline packages/ts-api/src/routes/example.ts:1]: Can this file be removed?

File: packages/ts-api/src/index.ts
Comment (erak) [inline packages/ts-api/src/index.ts:17]: Do we still need the example? (In line with my question below)

File: packages/ts-api/src/handlers/example.ts
Comment (erak) [inline packages/ts-api/src/handlers/example.ts:1]: Also remove this? Or do we need it as a template for upcoming features?

File: packages/ts-api/.example-env
Comment (erak) [inline packages/ts-api/.example-env:1]: I think this file should be renamed to `.pg-env` or `postgres-env` or something similar.

File: packages/ts-api/src/types/artists.ts
Comment (erak) [inline packages/ts-api/src/types/artists.ts:2]: Mmh, shall we somehow share these types between API and frontend? I makes totally sense to have 2 files as it is right now, because one lives in the frontend and one in the backend, but I think it could save us some effort during development if they would be one (no confusion about which file you're currently adding properties to, no missing properties etc.).

Comment (erak) [reply packages/ts-api/src/types/artists.ts:2]: If you're leaning towards keeping them separated I'd vote for using different types names.

**Review by erak — CHANGES_REQUESTED** [review]
So, in genreral, this looks really good! I found a few things, that I'd change before merging:

- remove example leftovers (`ts-api/src/handlers/example.ts` and such)
- maybe find a better way to seperate types used by the API and in the frontend? even an own package that cointains only the API types? Let's discuss!
- I have a strong opinion on meaningful commit messages. Maybe you can clean them up?

Thanks :-)

File: packages/ts-api/src/routes/example.ts
Comment (EV3RETH) [reply packages/ts-api/src/routes/example.ts:1]: oops, definitely!

File: packages/ts-api/.example-env
Comment (EV3RETH) [reply packages/ts-api/.example-env:1]: well this file will be renamed by new devs and turned into their actual .env file. I should put that in the readme though!

File: packages/ts-api/src/types/artists.ts
Comment (EV3RETH) [reply packages/ts-api/src/types/artists.ts:2]: I would like to as well. I did a quick search on sharing files between yarn workspaces and wasn't able to find anything helpful. If you know something or can find something I missed I would love to consolidate these into one folder.

**Comment by EV3RETH** [comment]
> So, in genreral, this looks really good! I found a few things, that I'd change before merging:
> 
> * remove example leftovers (`ts-api/src/handlers/example.ts` and such)
> * maybe find a better way to seperate types used by the API and in the frontend? even an own package that cointains only the API types? Let's discuss!
> * I have a strong opinion on meaningful commit messages. Maybe you can clean them up?
> 
> Thanks :-)

I hear you on the commit messages, in the future I will make the comments more meaningful!

File: packages/ts-api/src/handlers/example.ts
Comment (EV3RETH) [reply packages/ts-api/src/handlers/example.ts:1]: I was thinking of having the examples for future devs, but I suppose the files themselves will serve as that so we could remove them all

File: packages/ts-api/src/types/artists.ts
Comment (erak) [reply packages/ts-api/src/types/artists.ts:2]: What do you think about creating a shared workspace / package:

- packages/
  - next-frontend/
      - package.json
  - ts-api/
      - package.json
  - api-types/ (or similar)
      - src/
        - Types.tsx

Comment (erak) [reply packages/ts-api/src/types/artists.ts:2]: package.json could like this:
```
{
 ...
  "dependencies": {
    "api-types": "1.0.0",
  }
}

```

Comment (erak) [reply packages/ts-api/src/types/artists.ts:2]: Yeah, `types` sounds good.

File: package.json
Comment (erak) [inline package.json:16]: Indentation is off

File: packages/ts-api/src/types/artists.ts
Comment (EV3RETH) [reply packages/ts-api/src/types/artists.ts:2]: So as far as I can tell the best way to do this would be to publish an npm package with our types. On face value it doesn't seem that challenging but I'm inclined to hold off and make that its own ticket so that we can put some extra thought into it instead of rushing to do it to complete this ticket.

**Pull request merged**
